lib/spack/spack/cmd/create.py

```diff
@@ -184,14 +187,29 @@ class CMakePackageTemplate(PackageTemplate):
     """Provides appropriate overrides for CMake-based packages"""
 
     base_class_name = 'CMakePackage'
-
     body = """\
     def cmake_args(self):
         # FIXME: Add arguments other than
         # FIXME: CMAKE_INSTALL_PREFIX and CMAKE_BUILD_TYPE
         # FIXME: If not needed delete this function
         args = []
         return args"""
+    def __init__(self, name, url, versions):
+        PackageTemplate.__init__(self, name, url, versions)
+        with spack.stage.Stage(url, name) as variant_stage:
+            variant_stage.fetch()
+            variant_stage.expand_archive()
+
+            variant_stage.path
+            with spack.working_dir(variant_stage.source_path + "/build", create=True):
+                my_cmake = which('cmake')
+                my_cmake("..")
```

Comment:
This may be the not-so-good part about the PR.  It requires `CMake`.

We don't currently have a way for the regular Spack code to say "I depend on some cmake version being around" and build it, though we probably should for things like `curl` and the various version control tools.  That would not be ridiculously hard to implement.  But this is fine for a prototype -- I don't think you need to worry about that here.  For the moment, it should probably fail gracefully if `my_cmake` comes back `None`.